src/components/Rating.tsx

```diff
@@ -0,0 +1,58 @@
+import StarIcon from "@mui/icons-material/Star";
+import { Rating as RatingStars } from "@mui/material";
+import { SyntheticEvent } from "react";
+import { useRatedRecipe } from "../services/queries/useRatedRecipe";
+import { RatedRecipe } from "../types/documentTypes";
+
+interface RatingProps {
+  recipeId: string;
+  readOnly?: boolean;
+  userRating?: number;
+  ratingChangeHandler?: (
+    e: SyntheticEvent<Element, Event>,
+    value: number | null
+  ) => void;
+}
+
+const Rating: React.FC<RatingProps> = ({
+  readOnly = false,
+  recipeId,
+  userRating = 0,
+  ratingChangeHandler,
+}) => {
+  const { ratedRecipe } = useRatedRecipe("recipeId", recipeId);
+
+  const calculateAverageRating = (recipe: RatedRecipe[]): number => {
+    if (!recipe || recipe.length === 0) return 0;
+
+    const totalRecipeRating = recipe.reduce(
+      (sum, recipe) => sum + recipe.rating,
+      0
+    );
+
+    return totalRecipeRating / recipe.length;
+  };
+
+  const averageRating = ratedRecipe
+    ? calculateAverageRating(ratedRecipe as RatedRecipe[])
```

Comment:
Why do you need to write as RatedRecipe[]? it means the ratedRecipe is not typed well.